scripts/docker/Dockerfile

```diff
@@ -1,21 +1,79 @@
-FROM debian:9.8
+## LAYER base OS
+FROM debian:9.11 as os-base
 
 RUN rm /bin/sh && ln -s /bin/bash /bin/sh
 
+# Enable contributor package sources
+ARG DEBIAN_FRONTEND=noninteractive
 RUN sed -i 's/stretch main/stretch main contrib/' /etc/apt/sources.list
+RUN apt-get update && apt-get install --assume-yes apt-utils
 
-ENV DEBIAN_FRONTEND=noninteractive
+# General build niceties
+RUN apt-get update && apt-get install --assume-yes \
+	binutils \
+	bvi \
+	bzip2 \
+	coreutils \
+	git \
+	python3 \
+	rsync \
+	strace \
+	sudo \
+	tree \
+	unzip \
+	vim \
+	wget \
+	xz-utils \
+	zip \
+""
+
+# Install nibuild dependencies
+RUN dpkg --add-architecture i386
+RUN apt-get update && apt-get install --assume-yes \
+	libc6-i686:i386 \
+	libperl5.24:i386 \
+	libstdc++6:i386 \
+""
+RUN apt-get update && apt-get install --assume-yes \
+	build-essential \
+	ccache \
+	chrpath \
+	cvs \
+	desktop-file-utils \
+	diffstat \
+	g++ \
+	gawk \
+	help2man \
+	info \
+	libpulse-dev \
+	pkg-config \
+	subversion \
+	texi2html \
+	texinfo \
+	xmlstarlet \
+""
+
+# Install UTF-8 locales
+RUN apt-get update && apt-get install --assume-yes locales
+RUN dpkg-reconfigure -f noninteractive locales
+RUN echo 'en_US.UTF-8 UTF-8' > /etc/locale.gen && locale-gen en_US.UTF-8
+RUN update-locale LC_ALL=en_US.UTF-8 LANG=en_US.UTF-8
+ENV LANG='en_US.UTF-8' LANGUAGE='en_US:en' LC_ALL='en_US.UTF-8'
 
-RUN apt-get update && apt-get install --assume-yes apt-utils
+
+## LAYER build env
+FROM os-base as nilrt-build
 
 # Install OE dependencies
+ARG DEBIAN_FRONTEND=noninteractive
```

Comment:
Do you need this line repeated here?

Reply:
It failed without that, I suspect since I'm using ARG insted of ENV.  A few articles suggested ARG since that way any user of the container won't be stuck in noninteractive mode if they try to install anything, but I'm guessing it only applied for the current stage (and thus needs repeating at the next stage).

@amstewart let me know if I'm wrong here...